src/msix/msix.cpp

```diff
@@ -70,6 +70,25 @@ MSIX_API HRESULT STDMETHODCALLTYPE UnpackPackage(
     return static_cast<HRESULT>(MSIX::Error::OK);
 } CATCH_RETURN();
 
+MSIX_API HRESULT STDMETHODCALLTYPE UnpackPackageFromPackageReader(
+    IAppxPackageReader* packageReader,
+    MSIX_PACKUNPACK_OPTION packUnpackOptions,
+    char* utf8Destination) noexcept try
+{
+    ThrowErrorIfNot(MSIX::Error::InvalidParameter,
+        (packageReader != nullptr && utf8Destination != nullptr),
+        "Invalid parameters"
+    );
+
+    auto to = MSIX::ComPtr<IStorageObject>::Make<MSIX::DirectoryObject>(utf8Destination);
+
+    MSIX::ComPtr<IPackage> reader;
+    ThrowHrIfFailed(packageReader->QueryInterface(UuidOfImpl<IPackage>::iid, reinterpret_cast<void**>(&reader)));

```

Comment:
>QueryInterface [](start = 35, length = 14)

@rubengu@microsoft.com , do we not have a "From" on ComPtr?

Reply:
noup :)

---
In reply to: [291764778](https://github.com/Microsoft/msix-packaging/pull/152#discussion_r291764778) [](ancestors = 291764778)